gossip/gossip.go

```
@@ -311,6 +330,72 @@ func (g *GossipNodeSet) MergeRemoteState(buf []byte, join bool) {
 	}
 }
 
+// GossipEventReceiver is used to enable an application to receive
+// events about joins and leaves over a channel.
+//
+// Care must be taken that events are processed in a timely manner from
+// the channel, since this delegate will block until an event can be sent.
+type GossipEventReceiver struct {
+	ch           chan memberlist.NodeEvent
+	eventHandler pilosa.EventHandler
+}
+
+// NewGossipEventReceiver returns a new instance of GossipEventReceiver.
+func NewGossipEventReceiver() *GossipEventReceiver {
+	return &GossipEventReceiver{
+		ch: make(chan memberlist.NodeEvent, 1),
+	}
+}
+
+func (g *GossipEventReceiver) NotifyJoin(n *memberlist.Node) {
+	g.ch <- memberlist.NodeEvent{memberlist.NodeJoin, n}
+}
+
+func (g *GossipEventReceiver) NotifyLeave(n *memberlist.Node) {
+	g.ch <- memberlist.NodeEvent{memberlist.NodeLeave, n}
+}
+
+func (g *GossipEventReceiver) NotifyUpdate(n *memberlist.Node) {
+	g.ch <- memberlist.NodeEvent{memberlist.NodeUpdate, n}
+}
+
+// Start implements the pilosa.EventReceiver interface and sets the EventHandler.
+func (g *GossipEventReceiver) Start(h pilosa.EventHandler) error {
+	g.eventHandler = h
+	go g.listen()
+	return nil
+}
+
+func (g *GossipEventReceiver) listen() {
+	var nodeEventType pilosa.NodeEventType
+	for {
+		e := <-g.ch
+		switch e.Event {
+		case memberlist.NodeJoin:
+			nodeEventType = pilosa.NodeJoin
+		case memberlist.NodeLeave:
+			nodeEventType = pilosa.NodeLeave
+		case memberlist.NodeUpdate:
+			nodeEventType = pilosa.NodeUpdate
+		default:
+			continue
+		}
+
+		// Get the node from the event.Node meta data.
+		var n internal.Node
+		if err := proto.Unmarshal(e.Node.Meta, &n); err != nil {
+			panic("failed to unmarshal event node meta data")
+		}
+		node := pilosa.DecodeNode(&n)
+
+		ne := &pilosa.NodeEvent{
+			Event: nodeEventType,
+			Node:  node,
+		}
+		_ = g.eventHandler.ReceiveEvent(ne)
```

why is it ok to ignore this error?